treadmill/infra/connection.py

```diff
@@ -31,6 +31,9 @@ def __init__(self, resource=constants.EC2):
         pass
 
     def __new__(cls, resource=constants.EC2):
-        return boto3.client(
-            resource, region_name=cls.context.region_name
-        )
+        try:
+            return boto3.client(
+                resource, region_name=cls.context.region_name
+            )
+        except Exception as err:
+            print("AWS Client Exception: {}".format(str(err)))
```

Comment:
Not sure what the intent of this - logging? If logging, we do:

_LOGGER = logging.getLogger(__name__) in the beginning of the file and then:

_LOGGER.exception('aws client exception')

Reply:
This is a UI change intended to solve a setup issue I noticed when getting started with Treadmill. If the default AWS region is not defined in the user's environment, the connection is aborted and the user is faced with the enigmatic "Unable to load plugin: Cloud" error. 
This occurs before logging in cloud.py is initialized and before the new AWS exception handling is loaded. You showed me where to find a stack trace to debug the issue when I encountered this.

My intent with this commit was to sidestep that process and print the AWS connection error message for the user to diagnose. This has the benefit of delivering the error to the person who can correct it- we don't control the AWS configuration from within Treadmill. I'm happy to change the code to use the logger instead, but I believe we would then lose the UI benefits- the user would have to read the app logs to discover the issue.

Reply:
why isn't the exception (i am sure it is not "Exception" rather some high level exception) part of CLI_CLOUD_EXCEPTIONS list? 

the unable to load error happens on unhandled exceptions.... i do not see reason why can't we catch this properly.

I assume this code is used in cli and in long running processes. In this case, _LOGGER.debug or _LOGGER.exception is required, we do not print anything from anywhere.

I think it is best not to handle this at all at this level. For cli - it will be propagated and caught by exception handler, for sproc (long running) it will be unhanded and service will terminate as intended.